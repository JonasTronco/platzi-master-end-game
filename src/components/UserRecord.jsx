import PropTypes from 'prop-types';
import UserRecordOptions from './UserRecordOptions';

function UserRecord({ user }) {
  return (
    <tr className="table__body__row">
      <td className="table__body__row__cell__user">
        <img
          alt="user profile pic"
          src={
            user.imageURL ? user.imageURL : 'https://i.imgur.com/oMJFiLX.jpg'
          }
          className="table__body__row__cell__user__image"
        />
      </td>
      <td className="table__body__row__cell">
        <p>{user.username}</p>
      </td>
      <td className="table__body__row__cell">
        <p>{`${user.firstName} ${user.lastName}`}</p>
      </td>
      <td className="table__body__row__cell">
        <p>{user.typeOfUser}</p>
      </td>
      <UserRecordOptions active={user.isActive} />
    </tr>
  );
}

UserRecord.propTypes = {
  user: PropTypes.object.isRequired,
};

export default UserRecord;
