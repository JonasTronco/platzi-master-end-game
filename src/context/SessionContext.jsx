import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
