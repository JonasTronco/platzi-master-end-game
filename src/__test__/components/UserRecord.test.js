import UserRecord from '../../components/UserRecord';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('UserRecord Component', () => {
  test('Should render Login component', () => {
    const component = create(
      <SessionProvider>
        <UserRecord user={{}} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
