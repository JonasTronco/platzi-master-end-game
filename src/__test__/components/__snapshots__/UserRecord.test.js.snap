// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRecord Component Should render Login component 1`] = `
<tr
  className="table__body__row"
  onClick={[Function]}
>
  <td
    className="table__body__row__cell__user"
  >
    <img
      alt="user profile pic"
      className="table__body__row__cell__user__image--disabled"
      src="https://i.imgur.com/oMJFiLX.jpg"
    />
  </td>
  <td
    className="table__body__row__cell"
  >
    <span />
  </td>
  <td
    className="table__body__row__cell"
  >
    <span>
      undefined undefined
    </span>
  </td>
  <td
    className="table__body__row__cell"
  >
    <span />
  </td>
</tr>
`;
